telegram/_bot.py

```diff
@@ -442,13 +442,51 @@ async def _send_message(
         allow_sending_without_reply: ODVInput[bool] = DEFAULT_NONE,
         protect_content: ODVInput[bool] = DEFAULT_NONE,
         message_thread_id: int = None,
+        caption: str = None,
+        caption_entities: Union[List["MessageEntity"], Tuple["MessageEntity", ...]] = None,
         *,
         read_timeout: ODVInput[float] = DEFAULT_NONE,
         write_timeout: ODVInput[float] = DEFAULT_NONE,
         connect_timeout: ODVInput[float] = DEFAULT_NONE,
         pool_timeout: ODVInput[float] = DEFAULT_NONE,
         api_kwargs: JSONDict = None,
     ) -> Union[bool, Message]:
+        """Protected method to send messages of any type. Currently calls :meth:`_post`.
+
+        Note:
+            To reduce repetition (`if`-clauses) in public methods, the parameters
+            :paramref:`caption`, :paramref:`caption_entities`, :paramref:`message_thread_id`,
+            :paramref:`reply_markup`, and :paramref:`reply_to_message_id` are included
+            separately instead of them being passed in :paramref:`data`
+
+        Args:
+            endpoint (:obj:`str`): Telegram API endpoint
+            data (:class:`telegram._utils.types.JSONDict`): Data to be sent
+            reply_to_message_id (:obj:`int`, optional): |reply_to_msg_id|
+            disable_notification (:obj:`bool`, optional): |disable_notification|
+            reply_markup (:class:`ReplyMarkup`, optional): Additional interface options.
+                An object for an inline keyboard, custom reply keyboard, instructions to remove
+                reply keyboard or to force a reply from the user.
+            allow_sending_without_reply (:obj:`bool`, optional): |allow_sending_without_reply|
+            protect_content (:obj:`bool`, optional): |protect_content|
+
+                .. versionadded:: 13.10
+            message_thread_id (:obj:`int`, optional): |message_thread_id_arg|
+
+                .. versionadded:: 20.0
+            caption (:obj:`str`, optional): Media caption (may also be used when resending photos
+                by file_id), 0-:tg-const:`telegram.constants.MessageLimit.CAPTION_LENGTH`
+                characters after entities parsing.
+            caption_entities (List[:class:`telegram.MessageEntity`], optional): |caption_entities|
+
+        Returns:
+            :class:`telegram.Message`: On success, depending on a type of operation, the message
+                or :obj:`True` is returned.
+
+        Raises:
+            :class:`telegram.error.TelegramError`
+
+        """
         if reply_to_message_id is not None:
             data["reply_to_message_id"] = reply_to_message_id
```

Comment:
can you move this a few lines below so its a little more consistent

Reply:
Done